src/accessors/merge_reveal.rs

```diff
@@ -187,12 +187,12 @@ impl<Seal: ExposedSeal> MergeReveal for Assignments<Seal> {
 impl MergeReveal for TransitionBundle {
     fn merge_reveal(mut self, other: Self) -> Result<Self, MergeRevealError> {
         debug_assert_eq!(self.commitment_id(), other.commitment_id());
-        if self.known_transitions.len() + other.known_transitions.len() > self.input_map.len() ||
+        if self.input_map.len() < other.known_transitions.len() ||
```

Comment:
I think we need to take into account both `self` and `other` when do comparison. Why do you think the original version was buggy?

Reply:
All right, I see. I think it should be this way: first we do `extend`, than we do check

Reply:
I imagined that the original version made a purposeful comparison between `self` and `other`, so I adjusted it to a verification that made more sense.

So, should we first do the `extend` and then check if the amount of `input_map` is less than the `known_transitions` ?

Reply:
Yes

Reply:
Done